src/net/gateway.ts

```diff
@@ -32,14 +34,22 @@ export class Gateway {
      * Returns a new Gateway
      * @param dvoteGateway A DvoteGateway instance
      * @param web3Gateway A Web3Gateway instance
+     * @param params
      */
-    constructor(dvoteGateway: IDVoteGateway, web3Gateway: IWeb3Gateway) {
+    constructor(
+        dvoteGateway: IDVoteGateway,
+        web3Gateway: IWeb3Gateway,
+        params: { archiveIpnsId?: string, environment?: VocdoniEnvironment, networkId?: EthNetworkID } = {}
```

Comment:
Do we need environment and networkId?

Reply:
Shouldn't the IPNS ID have been defined on the Web3 gateway instead?
Why does the generic Gateway need to know the archive IPNS ID?